### ISSUE TITLE
Nextcloud 13.0.2 fails on Ubuntu 18.04 (when upgraded from 16.04)

### ISSUE THREAD

**Issue by holta** [iiab/iiab#754]
"./runrole nextcloud" doesn't seem to fully fix this, after changing /etc/iiab/iiab.env from `OS_VER=ubuntu-16` to `OS_VER=ubuntu-18` and running `./runrole nextcloud`, this error persists:
```
Error: PHP module cURL not installed [...if newly installed, please also...] restart the web server
```
Context: this failure's been confirmed on an Ubuntu 16.04 PC upgraded to 18.04 — also needs evaluation on a fresh install of Ubuntu 18.04 LTS

**Comment by holta** [comment]
> Error: PHP module cURL not installed [...if newly installed, please also...] restart the web server

@jvonau is helping trying to understand why this error persists even when php7.2-curl is confirmed installed:
```
root@box:~# apt -a list php7.2-curl
Listing... Done
php7.2-curl/bionic,now 7.2.3-1ubuntu1 amd64 [installed]
```
Clarif: "systemctl restart apache2" was run for good measure, but does not help.

**Comment by holta** [comment]
FYI after doing "mv /opt/nextcloud /opt/nextcloud.old" (as suggested at http://wiki.laptop.org/go/IIAB/FAQ#Can_I_upgrade_or_reinstall_server_apps.3F) "./runrole nextcloud" almost finishes but not quite:
```
TASK [nextcloud : Run Nextcloud initial install wizard] ************************
task path: /opt/iiab/iiab/roles/nextcloud/tasks/nextcloud_enabled.yml:42
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/commands/command.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1524925016.06-120419774510638 `" && echo ansible-tmp-1524925016.06-120419774510638="` echo /root/.ansible/tmp/ansible-tmp-1524925016.06-120419774510638 `" ) && sleep 0'
<127.0.0.1> PUT /root/.ansible/tmp/ansible-local-25865DlXeA3/tmpOUrY_A TO /root/.ansible/tmp/ansible-tmp-1524925016.06-120419774510638/command.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1524925016.06-120419774510638/ /root/.ansible/tmp/ansible-tmp-1524925016.06-120419774510638/command.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1524925016.06-120419774510638/command.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1524925016.06-120419774510638/ > /dev/null 2>&1 && sleep 0'
fatal: [127.0.0.1]: FAILED! => {
    "changed": true,
    "cmd": "cd /opt/nextcloud; sudo -u www-data php occ maintenance:install --database \"mysql\" --database-name \"nextcloud\" --database-user \"nextcloud\" --database-pass \"nextcloudmysql\" --admin-user \"Admin\" --admin-pass \"changeme\"",
    "delta": "0:00:00.365272",
    "end": "2018-04-28 10:16:56.554085",
    "invocation": {
        "module_args": {
            "_raw_params": "cd /opt/nextcloud; sudo -u www-data php occ maintenance:install --database \"mysql\" --database-name \"nextcloud\" --database-user \"nextcloud\" --database-pass \"nextcloudmysql\" --admin-user \"Admin\" --admin-pass \"changeme\"",
            "_uses_shell": true,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "warn": true
        }
    },
    "msg": "non-zero return code",
    "rc": 1,
    "start": "2018-04-28 10:16:56.188813",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "Nextcloud is not installed - only a limited number of commands are available\nThe username is already being used",
    "stdout_lines": [
        "Nextcloud is not installed - only a limited number of commands are available",
        "The username is already being used"
    ]
}
```

**Comment by holta** [comment]
FWIW all these things were attempted on the PC upgraded from Ubuntu 16.04/Desktop to 18.04/Desktop:
- systemctl disable appamor; reboot
- "a2enmod" then php7.2 to populate /etc/apache2/mods-enabled with php7.2.conf and php7.2.load
- "a2dismod" then php7.0 to remove php7.0.conf and php7.0.load from /etc/apache2/mods-enabled
- apt install php-curl
- apt purge php7.0-curl
- wipe prior Nextcloud installation as follows:
  - "rm -rf /opt/nextcloud"
  - "mysql -u root -p" then "DROP DATABASE nextcloud;"
  - "systemctl restart apache2" for good measure
- reinstall Nextcloud as follows:
  - "cd /opt/iiab/iiab; ./runrole nextcloud"

**Comment by holta** [comment]
FIXED!  (on PC that was upgraded from 16.04 to 18.04) -- the final clincher [above] to get http://box/nextcloud [and make error `PHP module cURL not installed`] was probably "a2dismod" then php7.0

**Comment by holta** [comment]
Ref PR #756 ("just to get initial run. not function tested (nextcloud)")

**Comment by holta** [comment]
Appears resolved, thanks to @georgejhunt's PR #756 which led to @jvonau 's PR #762 &mdash; but don't hesitate to re-open as necessary.
